insta-ui/src/app/server/llm/prompts.ts

```diff
@@ -0,0 +1,37 @@
+// app/server/llm/prompts.ts
+import { Message } from '@/lib/types'
+
+export function generateMessages(textInput: string, imageInput?: string, previousCode?: string): Message[] {
+    const messages: Message[] = [
+        {
+            role: 'system',
+            content: `You are a React component generator. Generate only valid, compilable React code. 
+      Follow these requirements:
+      - Use only React and standard packages
+      - Component must be compilable
+      - Include 'import React' at the start
+      - Use 'export default' for the component
+      - Ensure proper JSX syntax`
+        },
+        {
+            role: 'user',
+            content: `Create a React component based on this description: ${textInput}`
+        }
+    ];
+
+    if (imageInput) {
+        messages.push({
+            role: 'user',
+            content: `Include visual elements similar to this image: ${imageInput}`
+        });
+    }
```

Comment:
**_[This is trivial, we can fix this in later versions]_**

image input will need to be transformed to base64 format and wrapped up in a `ImageContent` object with `type: 'image_url'`.
<img width="726" alt="image" src="https://github.com/user-attachments/assets/feeea640-9cb0-4ad9-83bb-2ecae8bf6d0b">